pandas-stubs/core/algorithms.pyi

```diff
@@ -24,13 +40,13 @@ def unique(values: np.ndarray | list) -> np.ndarray: ...
 @overload
 def unique(values: ExtensionArray) -> ExtensionArray: ...
 def factorize(
-    values: Any,
+    values: Sequence | AnyArrayLike,
```

Comment:
Could improve this with an `overload` by saying that if `values` is `Categorical`, we return a `Categorical`.  If `Series` or `Index`, we return `Index`, o/w return `np.ndarray`

Reply:
I've added this.